onnxruntime/core/graph/graph_flatbuffers_utils.h

```diff
@@ -40,9 +40,39 @@ struct SparseTensor;
 
 namespace utils {
 
+/// <summary>
+/// Delegate to write initializer data to an external file/buffer.
+/// Data should be aligned to an appropriate boundary for the data type and or any potential mmap'd usage of the file.
+/// `data_type` is value returned by TensorProto::data_type() and is a value from onnx::TensorTypeProto_DataType.
+/// The function is not called for onnx::TensorTypeProto_DataType_STRING.
+/// The function should return the offset to the start of the data in the external file/buffer.
+/// </summary>
+using ExternalDataWriter = std::function<uint64_t(int32_t data_type, gsl::span<const uint8_t> bytes)>;
```

Comment:
nit: spans of `std::byte` instead of `uint8_t` may be better for representing data bytes.

Reply:
I think for this PR, we can leave them as spans of uint8_t because the UnpackTensor methods that the implementation relies on returns vectors / spans of uint8_t.

Reply:
ok that's fine